### ISSUE TITLE
Error: "TypeError: window.location.toString is not a function" 

### ISSUE THREAD

**Issue by jsnanigans** [DevExpress/testcafe-hammerhead#2178]
The issue has been reproduced here: 
https://github.com/jsnanigans/hammerhead_testcafe_error

### What is your Test Scenario?
Speculation: A iframe is focused, the iframe sends a message to the main window, the main window closes the iframe before testcafe can focus the main window.

> A Iframe is opened where the user enters some data and clicks on submit, then the iframe is redirected with 307 to where some javascript sends a postMessage to our host window which closes the iframe. After the iframe is closed and this error occurs something in the redux dispatch or in the react core breaks and the state is not updated/the app is not re-rendered.
> After some debugging we found that it might have something to do with hammerhead.js internals and some focus method.
> Important things to note here are that this happens if the iframe has a src (like in this example) or has no src attribute (like in our application)
 
### What is the Current behavior?
error is thrown and prevents react from rendering
 
### What is the Expected behavior?
the main window is focused in testcafe and continues as usual
 
### What is your web application and your TestCafe test code?
The issue has been reproduced here: 
https://github.com/jsnanigans/hammerhead_testcafe_error
 
### Steps to Reproduce:
1. clone repo: https://github.com/jsnanigans/hammerhead_testcafe_error
1. run npm install
1. run npm start to start the local server on port 3000.
1. run testcafe -c 1 chrome tests/Sample.ts --skip-js-errors to start testcafe.
1. when the test stops for debugging, open the console in the browser, there you will see the error.
 
### Your Environment details:
 
* testcafe version:                   1.6.1
* node.js version:                    v12.11.1
* command-line arguments:  `testcafe -c 1 chrome tests/Sample.ts --skip-js-errors`
* browser name and version:  Google Chrome Version 78.0.3904.87 (Official Build) (64-bit)
* platform and version:          macOS 10.14.6 (18G103)

**Comment by LavrovArtem** [comment]
I've researched this issue and created a simplified example. The problem occurs when the focus method is called on some element from a removed iframe.

```html
<body>
<script>
    var iframe = document.createElement('iframe');

    iframe.addEventListener('load', function () {
        var body = iframe.contentDocument.body;

        document.body.removeChild(iframe);
        body.focus();
    });

    document.body.appendChild(iframe);
</script>
```

```stack
VM2693 hammerhead.js:23480 Uncaught TypeError: window.location.toString is not a function
    at getLocation (VM2693 hammerhead.js:23480)
    at get (VM2693 hammerhead.js:23508)
    at Object.getParsed (VM2693 hammerhead.js:23523)
    at Function._wrapMessage (VM2693 hammerhead.js:13646)
    at MessageSandbox.sendServiceMsg (VM2693 hammerhead.js:13766)
    at ActiveWindowTracker.makeCurrentWindowActive (VM2693 hammerhead.js:11772)
    at raiseFocusEvent (VM2693 hammerhead.js:12664)
    at FocusBlurSandbox.focus (VM2693 hammerhead.js:12730)
    at HTMLBodyElement.focus (VM2693 hammerhead.js:13011)
    at HTMLIFrameElement.<anonymous> (iframe-focus.html:10)
```

**Comment by Farfurix** [comment]
https://github.com/DevExpress/testcafe-hammerhead/issues/2716#issuecomment-993247166

**Comment by nottoseethesun** [comment]
> #2716 (comment)

This fork I made:

  https://github.com/nottoseethesun/hammerhead_testcafe_error 

uses the latest `testcafe` (at the time of this writing) and has a couple features to make reproduction and analysis a lot more convenient (need to check the README though for a manual set-up step).

Proposed temporary workaround: Separately, to unblock myself I am going to try to workaround by wrapping the code where `location.toString` is invoked in the case where there is no [browsing context](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context), in a `try`-`catch` block, with a `console.warn`.  I think that will work as really, there seems to be nothing to be done by hammerhead anyway when an `iframe` has been removed.   The file in question is `./node_modules/testcafe-hammerhead/lib/client/hammerhead.min.js` and the function is `function re`, around line 254 under a Chrome DevTools pretty-print.  The patched file can be copied over at `npm`/`yarn` `postinstall`, which will be reliable as long as the `package.json` file pegs the version of `testcafe` (well, might need to peg `hammerhead` actually; we'll see).
